test/e2e/framework/resource_usage_gatherer.go

```diff
@@ -263,18 +265,37 @@ func NewResourceUsageGatherer(c *client.Client, options ResourceGathererOptions)
 			return nil, err
 		}
 
-		g.workerWg.Add(len(nodeList.Items))
-		for _, node := range nodeList.Items {
-			g.workers = append(g.workers, resourceGatherWorker{
-				c:                    c,
-				nodeName:             node.Name,
-				wg:                   &g.workerWg,
-				containerIDToNameMap: g.containerIDToNameMap,
-				containerIDs:         g.containerIDs,
-				stopCh:               g.stopCh,
-				finished:             false,
-				inKubemark:           false,
-			})
+		if options.masterOnly {
```

Comment:
Please merge it to a single for loop:

for _, node := range nodeList.Items {
  if !options.masterOnly || system.IsMasterNode(&node) {
    ...
  }
}